src/test/java/com/project/stockexchangeappbackend/service/OrderServiceImplTest.java

```diff
@@ -365,15 +367,17 @@ void shouldThrowAccessDeniedExceptionWhenDeactivatingOrder(
             @Mock SecurityContext securityContext, @Mock Authentication authentication) {
         Long id = 1L;
         Stock stock = createCustomStock(1L, "WIG20", "W20", 1024, BigDecimal.TEN);
-        User user = createCustomUser(1L, "test@test.pl", "John", "Kowal", BigDecimal.ZERO);
+        User user = createCustomUser(1L, "test@test.pl", "John", "Kowal", BigDecimal.ZERO, Role.USER);
+        User user2 = createCustomUser(2L, "test2@test.pl", "Timmy", "Lawok", BigDecimal.ZERO, Role.USER);
         Order order = createCustomOrder(id, 100, 100, OrderType.BUYING_ORDER, PriceType.EQUAL,
                 BigDecimal.TEN, OffsetDateTime.now().minusDays(1), OffsetDateTime.now().plusHours(2),
                 null, user, stock);
         SecurityContextHolder.setContext(securityContext);
 
         when(orderRepository.findById(id)).thenReturn(Optional.of(order));
+        when(userRepository.findByEmailIgnoreCase(user2.getEmail())).thenReturn(Optional.of(user2));
         when(securityContext.getAuthentication()).thenReturn(authentication);
-        when(authentication.getPrincipal()).thenReturn("user02");
+        when(authentication.getPrincipal()).thenReturn("test2@test.pl");
```

Comment:
You could use user2.getEmail() insted of "test2@test.pl". It's more flexible for changes.